### ISSUE TITLE
Game play for a typical turn

### ISSUE THREAD

**Issue by mclosson** [kungfucoder/Traveller#5]
@kungfucoder can you describe how a turn should work?  This will be very illustrative for what needs to be built in the software.  Here's some useful methods:

1. Assume no one else knows anything about how the game should work.  (A good assumption).
2. Give the game a starting point like "It is now player A's turn" and then repeatedly ask and answer the question "What happens next".  For example:

- It is player A's turn.  (What happens next?)
- Player A rolls the dice (What happens next?)
- Player A moves X spaces in any directions until the X moves are used (What happens next?)
- Player A is possibly presented with some situation on the map square in which he lands (What happens next?)
- Player A chooses an ability from an allowed list appropriate to the situation and uses it (What happens next?)
- A randomized dice roll occurs to determine the outcome which then... etc.

I'm not suggesting that this is the game play, only its an example of what is needed in order to communicate how the game should work and let all involved work towards it.

**Comment by kungfucoder** [comment]
Of course, I should have written a game flow out. This would help describe the actions carried out by the player. I'll try and write a basic version of it here, I'm on my phone at lunch.

Start
-Intro
-player defines his character
-player journeys to the first town
-Player is faced with a series of problems presented by the inhabitants.
-Player chooses a villager to help and solves the problem with an applicable skill. Ex:("you walk into town", "on your way in you see a reward for the capture of a local bandit holed up in the north.", "Further into town you notice an exasperated villager looking down at a broken wagon wheel." "There is also a lively tavern across the street." (Player chooses to help exasperated villager) "What an I going to do? I have to get these toys to the orphans in the next town over, or they won't have anything to live for." Choice of walking away or helping. (Player chooses to help) The player can either lift the wagon and let the villager put the wheel back on using strength, or appraise the wagon using craft to know more about the damage using craft. If player chooses strength right away and rolls successfully the wagon starts forward and then the wheel cracks and breaks. If the player appraises and is successful, the player knows it would take a craft to fix the wheel before it gets put on, which could be done with another successful craft check. (Player chooses appraise and succeeds, then succeeds on repair) "wagon starts forward and continues on it's journey to deliver the toys to the sad orphans."  The player is then prompted for their next action, which could be going into the bar, going to the bounty office, or exploring more of the town. Once you explore the town fully and find all the options for you to investigate, the player can either move on to the next town or solve some or all of the problems in the town. 
There's more that would be easier to put in this flow if I were on the PC, but that's basically what I was thinking. Less combat, and more interacting with people.

**Comment by kungfucoder** [comment]
-Start
-Introduction
-Player inputs information into character class
-player journeys to town
-player is introduced to a problem in the town

-  if player accepts problem
    -player will decide skill check to use and roll to see if it is successful or a failure

-  if player bypasses the problem
    -player will move on in the game and be introduced to another problem in the town
-repeat process until all problems are either bypassed or solved
    -success, failure and, bypassing may have an impact on problems introduced in towns not yet visited

-player moves onto the next town and is introduced to more complex problems, involving the requirement to interact with certain people in the town to find specific clues, or items.
    -some of which may only be able to be acquired by successfully solving a problem that was presented 
     in the previous town

**Comment by kungfucoder** [comment]
I'll try and illustrate a better example than the rushed wagon wheel I gave earlier.

Ex: 
-Player enters town and sees a small wooden hut starting to catch fire
"A woman comes running out and screams 'HELP! My house is on fire!'"
"Townsfolk have begun securing their own homes, looks like this problem may be up to you."
"What do you do?"
(options given to player)
  -Use strength to carry a small water barrel you see near you to the house and attempt to douse the 
    flames
  -Use charisma to try and persuade the townspeople to help put out the fire before it spreads
  -Let it burn

[outcome for each option]
-player chooses to use strength to carry the water barrel and attempt to douse the flames
   -player rolls dice and succeeds
      -"the fire is quenched and the house is saved. The woman is so grateful she gives you half of her life
        savings."
   -player rolls dice and fails
       -"You carry the barrel over, but begin to tire and the fire spreads and burns the house down."
-Player attempts to persuade the townsfolk to help put out the fire before it spreads
   -player rolls dice and succeeds
       -"the fire is put out by the efforts of the community, the woman is so thankful to you for convincing   
          the town to help she gives you half her life savings."
   -player rolls dice and fails
        -"You attempt to give an impassioned speech about community, but the smoke from the growing 
           sends you into a coughing fit. The house is burnt to the ground."
-Player lets it burn
    -"You stand there watching the flames gradually consume the house and reduce it to ash."
        -The woman is sobbing in front of where her house used to be.
    (Options if house is not saved)
-You can use charisma to console the woman and convince he.r that at least she has her life.
-You can tell the woman you will help her rebuild and use craft to build her a new home.
-You can walk away.
